drivers/usb/usbport/usb2.c

```diff
@@ -7,22 +7,2236 @@
+        Result = FALSE;
+    }
+
+    return Result;
+}
+
+USHORT
+NTAPI
+USB2_AddDataBitStuff(IN USHORT DataTime)
+{
+    return (DataTime + (DataTime / 16));
+}
+
+VOID
+NTAPI
+USB2_IncMicroFrame(OUT PUCHAR frame,
+                   OUT PUCHAR uframe)
+{
+    ++*uframe;
+
+    if (*uframe > (USB2_MICROFRAMES - 1))
+    {
+        *uframe = 0;
+        *frame = (*frame + 1) & (USB2_FRAMES - 1);
+    }
+}
+
+VOID
+NTAPI
+USB2_GetPrevMicroFrame(OUT PUCHAR frame,
+                       OUT PUCHAR uframe)
+{
+    *uframe = USB2_MICROFRAMES - 1;
+
+    if (*frame)
+        --*frame;
+    else
+        *frame = USB2_FRAMES - 1;
+}
+
+BOOLEAN
+NTAPI
+USB2_CheckTtEndpointInsert(IN PUSB2_TT_ENDPOINT nextTtEndpoint,
+                           IN PUSB2_TT_ENDPOINT TtEndpoint)
+{
+    ULONG TransferType;
+
+    DPRINT("USB2_CheckTtEndpointInsert: nextTtEndpoint - %p, TtEndpoint - %p\n",
+           nextTtEndpoint,
+           TtEndpoint);
+
+    ASSERT(TtEndpoint);
+
+    if (TtEndpoint->CalcBusTime >= (USB2_FS_MAX_PERIODIC_ALLOCATION / 2))
+    {
+        DPRINT1("USB2_CheckTtEndpointInsert: Result - FALSE\n");
+        return FALSE;
+    }
+
+    if (!nextTtEndpoint)
+    {
+        DPRINT("USB2_CheckTtEndpointInsert: Result - TRUE\n");
+        return TRUE;
+    }
+
+    TransferType = TtEndpoint->TtEndpointParams.TransferType;
+
+    if (nextTtEndpoint->ActualPeriod < TtEndpoint->ActualPeriod &&
+        TransferType == USBPORT_TRANSFER_TYPE_INTERRUPT)
+    {
+        DPRINT("USB2_CheckTtEndpointInsert: Result - TRUE\n");
+        return TRUE;
+    }
+
+    if ((nextTtEndpoint->ActualPeriod <= TtEndpoint->ActualPeriod &&
+        TransferType == USBPORT_TRANSFER_TYPE_ISOCHRONOUS) ||
+        nextTtEndpoint == TtEndpoint)
+    {
+        DPRINT("USB2_CheckTtEndpointInsert: Result - TRUE\n");
+        return TRUE;
+    }
+
+    DPRINT("USB2_CheckTtEndpointInsert: Result - FALSE\n");
+    return FALSE;
+}
+
+ULONG
+NTAPI
+USB2_GetOverhead(IN PUSB2_TT_ENDPOINT TtEndpoint)
+{
+    ULONG TransferType;
+    ULONG Direction;
+    ULONG DeviceSpeed;
+    ULONG Overhead;
+    ULONG HostDelay;
+
+    TransferType = TtEndpoint->TtEndpointParams.TransferType;
+    Direction = TtEndpoint->TtEndpointParams.Direction;
+    DeviceSpeed = TtEndpoint->TtEndpointParams.DeviceSpeed;
+
+    HostDelay = TtEndpoint->Tt->HcExtension->HcDelayTime;
+
+    if (DeviceSpeed == UsbHighSpeed)
+    {
+        if (Direction == USBPORT_TRANSFER_DIRECTION_OUT)
+        {
+            if (TransferType == USBPORT_TRANSFER_TYPE_ISOCHRONOUS)
+                Overhead = HostDelay + USB2_HS_ISOCHRONOUS_OUT_OVERHEAD;
+            else
+                Overhead = HostDelay + USB2_HS_INTERRUPT_OUT_OVERHEAD;
+        }
+        else
+        {
+            if (TransferType == USBPORT_TRANSFER_TYPE_ISOCHRONOUS)
+                Overhead = HostDelay + USB2_HS_ISOCHRONOUS_IN_OVERHEAD;
+            else
+                Overhead = HostDelay + USB2_HS_ISOCHRONOUS_OUT_OVERHEAD;
```

Comment:
Shouldn't this be USB2_HS_INTERRUPT_IN_OVERHEAD?

Reply:
Yes, fixed.